scripts/bootstrap

```diff
@@ -22,6 +22,18 @@ echo "using rust version '$rust_version'"
 # here we set the toolchain to 'none' and rustup will pick up on ./rust-toolchain.toml
 run curl --fail https://sh.rustup.rs -sSf | run sh -s -- -y --default-toolchain "none" --no-modify-path
 
+# https://blog.rust-lang.org/2025/03/02/Rustup-1.28.0.html
+# Check if the required toolchain is installed; if not, install it
+if ! rustup toolchain list | grep -q "$rust_version"; then
```

Comment:
is the check necessary? wouldn't `rustup toolchain install ...` simply be a no-op if the toolchain is already installed?

Reply:
Yes, I'm just a nooby for now. I thought that this would do no harm when I saw it suggested somewhere and decided to add it.

We'll improve this to set the rustup version, so I'll remove this check.